### DISCUSSION TITLE
Resistor values for the LEDs

### DISCUSSION THREAD

**Discussion by jimbotel** [tjko/fanpico#13] [Q&A] [answered]
Hi @tjko ,

Could you tell me how did you come up with the 49.9K ohm value for the resistors accompanying the LEDs for the 12v voltage ?
They seem to be really high values to me.
(Same for the 10K ohm for 5V and 4.99K ohm for 3.3V)

Thanks!
Jaime

**Comment by tjko** [comment, marked as answer]
LEDs used are "high-intensity" ones, around rated current (10mA if I recall correctly) they were way too bright....

I picked the resistors so that the LEDs didn't look "too bright" by just eyeballing while adjusting resistance using a resistance decade box... Then I picked resistor values based on something reasonably close that I happened to have...

Brightnesses look pretty close by just looking at the LEDs, but if doing math (and trying to match that ~ 240uA for the 12V LEDs), then probably 20k for 5V and 13.7k for 3.3V would produce more uniform brightness....

**Reply by jimbotel** [reply]
hi @tjko , 
Thanks a lot for your prompt reply, I appreciate the answers. FYI, this summer I started to create a kind of "fanpico" board -I was not aware of your project at that time- and I'd like to use some of your ideas on it, like the LEDs. For me the project goal is to refresh a closed compartment I have behind the TV where I have all the electronic stuff (router, switch, UPS, NAS, other raspberry pi-s, etc), so the idea was to place two fans at the bottom of the front door introducing air from outside on it and two other fans at the top getting hot air out, controlling the speed (duty cycle) of the fans depending on the temperature (using a DHT22 sensor). I also added a relay to stop powering the fans if temperature is below a threshold, an OLED to display info (same as you did with the "D" version) and a buzzer to alert in case some problem is found (e.g. if a fan stop working). Some of these components are "external" (I purchased separate small boards, like power conversion or relay) to avoid over-complicating things -see the other goals below-. I also plan to extend the project by sending metrics information to another raspberry pi board -hence the pico "w"- so I can analyze and present the data using a frontend built with Node-RED. Most of these ideas are related to the second goal of teaching.
Second and third goals of my project are:
  - Introduce some students (friends of my son) into MCUs and Python programming, so everything is written in Python.
  - Learn KiCad (and teach to the students) : I've never used it before, I'm taking a course in Udemy and I wanted a real project to start from scratch and apply the knowledge into.
  - Learn to work with SMD components, also a new field to me.
  
 It was a real coincidence when I saw your project, even more that you are also using the raspberry pi pico. Initially I thought about using an ESP32 MCU which I believe is ahead of the rpi pico in terms of features and even price, but I thought the pico is probably better suited for teaching, so I chose that. 
In case you are interested, my on-going work is here:
Programming stuff: https://github.com/jimbotel/rpi_pico_projects  ("main.py", that's the important one -not sure if it's up-to-date though-) 
KiCad design:  https://github.com/jimbotel/fan_picow_kicad (probably not the latest of the latest version, but you get the idea)
It's a newbie work, nothing professional, as I mentioned it is a "learning experience" overall :-).

Thanks,
Jaime-

**Reply by tjko** [reply]
Hi @jimbotel, 
Your project looks interesting. I like the idea of having a buzzer for audible warning, etc...  
Btw, some (high end) PC fans can be stopped with low duty-cycle PWM signal (Noctua, etc...) so you might not necessarily need relays. And one option would be to use MOSFET as "relay" instead...
Feel free to use anything you may find useful (and everything is released under GPL anyways). Just keep in mind "fanpico" is also a hobbyist project...

I also looked ESP32 boards (and Arduinos), but Pico and Pico W seemed much more flexible (and provide all kinds of features like PIO that other MCUs don't have at this price point...).

**Reply by jimbotel** [reply]
hi @tjko ,
Thanks for your comments, they are very much appreciated.
I purchased a box with five Arctic F8 PWM PST fans which by design have a minimum speed (>0) when the duty cycle is 0%. I don't think this is because they are "not high-end", but because they decided to implement them that way (could be a "PC protection measure").  See:
 https://www.arctic.de/en/F8-PWM-PST/ACFAN00064A
 https://www.arctic.de/media/f9/28/58/1654783278/Spec_Sheet_F8_PWM_PST_EN.pdf
I agree with you that using a MOSFET instead of a relay could be a better idea. To be frank, I reviewed both solutions before taking the decision of using a relay; I had the idea (probably a missconception) that the MOSFET would drain some current all of the time when it is "on", while the relay would do it only the time you switch it. In my use case, probably the fans are always on for months (spring, summer, autumn) and then off the rest of the time (winter), so switching on/off is a "rare" case. I want a solution that drains very little current while it is on (and if possible while it is off).
Also, I followed this documentation when dealing with the fans:
https://www.intel.com/content/dam/support/us/en/documents/intel-nuc/intel-4wire-pwm-fans-specs.pdf
which seems to be the "official" specs for PC fans. For the tachometer signal I followed this schematic (the "resistors path" 4.7K / 27.4K / 10K ):
  https://www.electroschematics.com/wp-content/uploads/2019/12/2-Fan-Connector-Scheme.png
I tested that and it worked fine for me.
I'm curious, where did you get your implementation for the tachometer signal circuit from?
Your project is fantastic, if I was not looking at "starting from scratch" I would go with that, but right now I want to learn by doing (make my own mistakes), including sending my PCB to the manufacturer, programming everything -in my case in Python-, etc.

**Reply by tjko** [reply]
Hi @jimbotel ,

I sort of came up with the tachometer signal circuit myself.
I also started from the Intel "spec", and based on it saying fans must have "open collector" output, I initially just had a pull-up resistor on a GPIO pin (based on some example how to read open collector outputs with a MCU).  Things seemed to work fine, until I came across a fan that had 12V internal pull up on tachometer signal (which is against Intel specs!), and I was seeing 12V at the input pin, not good... So I ended up adding diode to isolate (or protect) MCU from any pull-up voltage fan might have on the tachometer signal. 

Later, when testing the circuit on breadboard, I noticed that the tachometer signal was somewhat noisy (when lookin on oscilloscope screen). So, I ended up adding small capacitor to make signal much 'cleaner'....  
MCU didn't seem to mind the noisy signal, but sure won't hurt to add inexpensive capacitor (at least signal now looks clean if looking on a scope).

**Reply by jimbotel** [reply]
Hi @tjko ,
Thanks again for the answers, I learn a lot from them. These weeks I plan to do additional tests adding some of your ideas and I'll make changes to my design based on them. BTW I have a working prototype running since mid September and it seems to be fine..
I created this album with my working prototype setup: 
    https://1drv.ms/u/s!Ap406dcIZZtYgdot2vNUB0zFZckNvg?e=Cn0lPe


Regards,
Jaime
